cpp/benchmarks/pairwise_linestring_distance.cu

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+#include <nvbench/nvbench.cuh>
+
+#include <cuspatial/experimental/linestring_distance.cuh>
+#include <cuspatial/experimental/type_utils.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <memory>
+
+namespace cuspatial {
+
+/**
+ * @brief Helper to generate linestrings used for benchmarks.
+ *
+ * The generator adopts a walking algorithm. The ith point is computed by
+ * walking (cos(i) * segment_length, sin(i) * segment_length) from the `i-1`
```

Comment:
What is the goal for the shape of the linestrings? cos(i) and sin(i) are non-intuitive for me when `i` is an integer > 1.  Is this just a way to get a random value in [0,1]?

Reply:
Yup, it's more like a random walk algo here. The shape doesn't really matter but the locality of the linestring is more important, since the pair needs to separate far from each other to take the long code path to compute the distance.

Reply:
Can you explain this in the comment?